### PR TITLE
Add guide on handling tool calling errors

### PR DESCRIPTION
**Pull request by jacoblee93**

### CODE REVIEW

**Comment by vbarda** [comment]
thanks for adding! looks good overall, 2 comments:

* i would probably use the same example in both cases for consistency, otherwise feels a bit disconnected
* for the last example, can we use streaming instead, to show what was happening? i know we have a langsmith trace, but people might ignore that and it's not-obvious from the notebook outputs

File: examples/tool-calling-errors.ipynb
Comment (hinthornw) [inline examples/tool-calling-errors.ipynb:1]: "Calling tools with an LLM is generally more reliable than pure prompting, but it isn't perfect" idk if this take is common, but I don't typically think of tools as a replacement for pure prompting. They're just the way the LLM interfaces with the rest of your program and with the outside world.

Comment (jacoblee93) [reply examples/tool-calling-errors.ipynb:1]: What would you suggest?

**Comment by jacoblee93** [comment]
> thanks for adding! looks good overall, 2 comments:
> 
> * i would probably use the same example in both cases for consistency, otherwise feels a bit disconnected
> * for the last example, can we use streaming instead, to show what was happening? i know we have a langsmith trace, but people might ignore that and it's not-obvious from the notebook outputs

1. They're two separate failure modes - I guess I could try to say "generate a haiku about a specific place" or three different cities but it'd be tough to prompt engineer.
2. Done

**Pull request merged**